sklearn/linear_model/_ridge.py

```diff
@@ -2234,10 +2290,28 @@ def fit(self, X, y, sample_weight=None):
 
         return self
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
 
-class RidgeCV(
-    _RoutingNotSupportedMixin, MultiOutputMixin, RegressorMixin, _BaseRidgeCV
-):
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.4
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = MetadataRouter(owner=self.__class__.__name__).add(
```

Comment:
this estimator in effect is a consumer of `sample_weight`, so we need to add `self`